ccf_transaction_processor/transaction_processor/pdo_tp.cpp

```diff
@@ -0,0 +1,624 @@
+                    jsonrpc::StandardErrorCodes::INVALID_PARAMS, "Contract has been turned inactive. No more upates permitted");
+            }
+
+            // ensure that the enclave is part of the contract (no need to separately check if enclave is registered)
+            bool is_enclave_in_contract = false;
+            for (auto enclave_in_contract: contract_info.enclave_info){
+                if (in.contract_enclave_id == enclave_in_contract.contract_enclave_id) {
+                    is_enclave_in_contract = true;
+                    break;
+                }
+            }
+            if (!is_enclave_in_contract) {
+                return make_error(
+                        jsonrpc::StandardErrorCodes::INVALID_PARAMS, "Enclave used for state update not part of contract");
+            }
+
+            if (in.verb == "init") {
+            // Ensure the following:
+            // 1. no previous updates were ever performed, 2. no previous state hash in update_info 3. no depedency list
+            // 4. ensure that the update operation is performed by the contract owner 5. verify init signature
+
+                if (contract_info.current_state_hash.size() != 0){
+                    return make_error(
+                        jsonrpc::StandardErrorCodes::INVALID_PARAMS, "Contract has already been initialized");
+                }
+
+                if (state_update_info.previous_state_hash.size() != 0){
+                    return make_error(
+                        jsonrpc::StandardErrorCodes::INVALID_PARAMS, "Init operation must not have any previous state hash");
+                }
+
+                if (state_update_info.dependency_list.size() != 0){
+                    return make_error(
+                        jsonrpc::StandardErrorCodes::INVALID_PARAMS, "Init operation must not have CCL dependeices from other contracts");
+                }
+
+                if (in.verifying_key != contract_info.contract_creator_verifying_key_PEM){
+                    return make_error(
+                        jsonrpc::StandardErrorCodes::INVALID_PARAMS, "Init operation can only be performed by the contract creator");
+                }
+
+                if (!verify_pdo_transaction_signature_update_contract_state(in.signature, in.verifying_key, \
+                    in.contract_enclave_id, in.contract_enclave_signature, in.state_update_info)){
+                    return make_error(
+                    jsonrpc::StandardErrorCodes::INVALID_PARAMS, "Invalid PDO payload signature");
+                }
+
+            }   else if (in.verb == "update") {
+            // Ensure the following:
+            // 1. the previous state hash (from incoming data) is the latest state hash known to CCF (this also ensures that
+            //                there was an init)
+            // 2. depedencies are met (meaning these transactions have been committed in the past)
+            // 3. there is a change in state, else nothing to commit
```

Comment:
Thank you. As you rightly say the check for "submitter can only invoke the cotnract" is part of the enclave sign check. I will put a comment about this just before the sign check. (since this applies to both init & update)